languages/sqlite/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/kr/pty"
+	"golang.org/x/sys/unix"
+
+	"github.com/replit/prybar/utils"
+)
+
+var Instance = &SQLite{}
+
+type SQLite struct {}
+
+// constructConfigFile generates commands to configure the sqlite CLI.
+// It writes them to a temporary file and returns its pathname.
+func constructConfigFile(config *utils.Config) string {
+	f, err := ioutil.TempFile("", "sqlite-config")
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+
+	// main and continuation prompts
+	// TODO: this probably doesn't handle quotation marks properly
+	f.WriteString(fmt.Sprintf(".prompt '%s' '%s'\n", config.Ps1, config.Ps2))
+
+	return f.Name()
+}
+
+func disableEcho(file *os.File) {
+	fd := int(file.Fd())
+	termios, err := unix.IoctlGetTermios(fd, unix.TCGETS)
+	if err != nil {
+		panic(err)
+	}
+
+	termios.Lflag &^= unix.ECHO
+
+	if err := unix.IoctlSetTermios(fd, unix.TCSETS, termios); err != nil {
+		panic(err)
+	}
+}
+
+func Execute(config *utils.Config) {
+	configFile := constructConfigFile(config)
+	args := []string{"-init", configFile}
+	cmd := exec.Command("sqlite3", args...)
+
+	ptty, tty, err := pty.Open()
```

Comment:
the prybar process is already opened under a pty (i.e. stdin/stdout/stderr are all FDs to a pty from pid1) so just use that (they're available as os.Stdin/os.Stdout/etc).